src/views/AddItem.jsx

```diff
@@ -29,19 +32,42 @@ export function AddItem({ listToken }) {
 
 	// Store values inside itemData and sent over to database
 	// otherwise log an error if request fails
-	const onFormSubmit = (event) => {
-		event.preventDefault();
+	const onFormSubmit = (e) => {
+		e.preventDefault();
 
 		// if list token is not provided, notify user and exit function
 		if (!listToken) {
 			notifyNoToken();
 			return;
 		}
 
+		// We are looking for an array of items that matches the user input
+		// We use `toLowerCase` and `replaceAll` to account for edge cases,
+		// such as casing and extra spaces
+		const nameMatchesArr = data.some((item) => {
+			return (
+				item.name &&
+				item.name.replaceAll(' ', '') ===
+					itemName.toLowerCase().replaceAll(' ', '')
+			);
+		});
+
+		// The user is notified if the input matches an item in db
+		// Item does not get added; exits the function
+		if (nameMatchesArr) {
+			notifyInList();
+			return;
+		}
+
+		// Updates the relevant data in db
 		const itemData = {
 			itemName,
 			daysUntilNextPurchase,
 		};
+
+		// If the input has no match in db, the item is added, notifies the user,
+		// and sets input back to empty string.
+		// Errors are handled otherwise.
```

Comment:
I really appreciate you both putting in additional comments throughout `AddItem.js` to make the code more readable. I've found myself having some trouble looking back at previous weeks code that didn't have explicit comments above it and am happy to see y'all making these easier to absorb!

Reply:
+1!